### ISSUE TITLE
"delete dumpstat/logcat"

### ISSUE THREAD

**Issue by lemmy04** [d4rken/sdmaid-public#5188]
**Is your feature request related to a problem? Please describe.**
general performance issues with android phones... 

**Describe the solution you'd like**
There is a "hack" being published on the internet for performance issues with android devices:
- dial *#9900# to open a "hidden" menu
- in that menu, tap "delete dumpstate/logcat"

The result is a much more responsive device, bordering on miraculous.

I've personally tested this with four different samsung devices - no bad sideeffects but awesome results.

could this become part of the "system cleaner" part of SD maid?

**Comment by lemmy04** [comment]
after a bit of digging:

the data that gets deleted by this is in /dats/log so it would need root after all - and the *#9900# shortcut only works on some but not all samsung devices.

**Comment by lemmy04** [comment]
more digging: what the function does is (in shell code):

```
find /data/log -type f -exec rm {} \;
rm -rf /data/log/acore
```

**Comment by d4rken** [comment]
Can you provide a path dump of that folder, e.g. SD Maids explorer option, or some other way to show me the full file structure/permissions/content of the location?

**Comment by lemmy04** [comment]
i just cleaned it on all my devices - let me give them a few days of hard use to get it "nice and filthy" in there, then i'll just post a "ls -lR" before and after cleaning (my phones all have sshd running...)

**Comment by lemmy04** [comment]
so here is a "ls -lR /data/log" before cleaning it in that hidden menu:

https://paste.opensuse.org/59778340

and one **after** cleaning it out:

https://paste.opensuse.org/11046461

**Comment by lemmy04** [comment]
Any progress on this?

**Comment by d4rken** [comment]
Sorry, no

**Comment by lemmy04** [comment]
can I help somehow?

**Comment by d4rken** [comment]
Not really, any time I can allocate time, it keeps getting burned away by regressions in newer Android versions. If I work on something it is also often something with a lot of edge-cases that ends up breaking for some users, while fixing it for others, being a huge headache 😦. 

SD Maids architecture was pretty nice allowing quick iterations around Android 10, but Google changes too many things that I didn't expect to be changed and where my architecture falls a bit short.

I'm contemplating a rewrite by creating new SD Maid, rewriting the current app is bug prone and surely pisses off the existing users that are happy with the way SD Maid performs (or looks).

I'd focused on the most used features, e.g. CorpseFinder, SystemCleaner and AppCleaner. No need to compete with others apps on being an "Explorer" app. We'd end up with less features, but with SD Maid being better at the remaining features, and less overhead for bugfixes and features (within scope). 🤔 

What do you think?

**Comment by lemmy04** [comment]
sounds about right to me... no need to follow this silly trend where any app no matter what it does eventually also contains a built in web browser, email client, messenger and root file explorer.

But then I'm coming from 26 years of unix, where every tool does only ONE thing but does it right :)
